github-actions/rebase-pr/entrypoint.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+set -e
+
+# Set email and name config
+git config --global user.email "devops+rebase@angular.io"
+git config --global user.name "Angular Rebase Bot"
+
+# Exit for code a successful run
+SUCCESS_EXIT_CODE=0
+
+# The repository name and branch name for the PRs head and base
+HEAD_REPOSITORY=$(jq -r ".pull_request.head.repo.full_name" "$GITHUB_EVENT_PATH")
+HEAD_BRANCH=$(jq -r ".pull_request.head.ref" "$GITHUB_EVENT_PATH")
+BASE_REPOSITORY=$(jq -r ".pull_request.base.repo.full_name" "$GITHUB_EVENT_PATH")
+BASE_BRANCH=$(jq -r ".pull_request.base.ref" "$GITHUB_EVENT_PATH")
+
+# Change to the github workspace and initialize the repo if needed.
+cd "$GITHUB_WORKSPACE"
+git init
+
+# Add head and base repos as remotes, then fetch the remote branches
+git remote add head https://x-access-token:$TOKEN@github.com/$HEAD_REPOSITORY
+git remote add base https://x-access-token:$TOKEN@github.com/$BASE_REPOSITORY
```

Comment:
Is this the recommended way to use the token? Is there any chance this could result in the token's showing up in some logs or something?

Reply:
Github prevents the printing of the token in the logs, so it shouldn't be a problem.

I agree that I don't love it as the method for providing the token, I was unfortunately not able to find another way that worked in this context to provide the token to git.

Reply:
You can probably write something to a file and set that file as the credential helper store. Something similar to what we do on [CircleCI][1] (without the decryption step). Plus you might also need some `git config credential.helper "store --file=/path/to/credentials/file"`.

But that is not bullet-proof either. Not worth the extra work if GitHub prevents printing the token in the logs.

[1]: https://github.com/angular/angular/blob/64823f561c2c25f05e9bce04de68374ce997548a/.circleci/config.yml#L580